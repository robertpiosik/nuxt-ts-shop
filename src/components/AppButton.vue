<template lang="pug">
	component(
		:is="buttonType"
		:to="to"
		:href="href"
		v-on="$listeners"
		:class="$style.button"
		v-bind="$attrs"
	)
		slot
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'vue-property-decorator'

@Component
export default class extends Vue {
  @Prop(String) readonly to?: string
  @Prop(String) readonly href?: string
  @Prop({ type: String, default: 'normal' }) readonly size!:
    | 'small'
    | 'normal'
    | 'large'
  @Prop({ type: Boolean, default: false }) readonly isLoading!: boolean

  get buttonType() {
    if (this.to && this.to.length > 0) {
      return 'nuxt-link'
    } else if (this.href && this.href.length > 0) {
      return 'a'
    } else {
      return 'button'
    }
  }
}
</script>

<style lang="scss" module>
.button {
  @include toExtraSmall {
    font-size: 1.4rem;
    padding: 0.6rem 1rem;
  }
  background: $primary;
  color: white;
  font-weight: 600;
  padding: 1rem 1.3rem;
  border-radius: 0.5rem;
  &:hover {
    filter: brightness(120%);
  }
  &:active {
    transform: scale(0.98);
  }
}
</style>
