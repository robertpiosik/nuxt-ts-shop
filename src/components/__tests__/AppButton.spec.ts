import { mount } from '@vue/test-utils'
import AppButton from './../AppButton.vue'

const wrapper = mount(AppButton)

describe('Component: AppButton', () => {
  describe('GENERAL', () => {
    it('should match a snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
    it('should be a Vue instance', () => {
      expect(wrapper.isVueInstance()).toBeTruthy()
    })
  })
})
