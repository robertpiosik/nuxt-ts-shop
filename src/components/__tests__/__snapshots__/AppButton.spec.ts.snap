// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AppButton GENERAL should match a snapshot 1`] = `<button class="button"></button>`;
